miner/algo_common.go

```diff
@@ -281,15 +296,28 @@ func (envDiff *environmentDiff) commitBundle(bundle *types.SimulatedBundle, chDa
 			return errInterrupt
 		}
 
+		// update isRevertibleTx if transaction is found in list of reverting transaction hashes
+		isRevertibleTx = bundle.OriginalBundle.RevertingHash(txHash)
+
 		receipt, _, err := tmpEnvDiff.commitTx(tx, chData)
 
-		if err != nil {
-			log.Trace("Bundle tx error", "bundle", bundle.OriginalBundle.Hash, "tx", tx.Hash(), "err", err)
+		hasCommitErr = err != nil
+		hasReceiptFailed = receipt != nil && receipt.Status == types.ReceiptStatusFailed
+
+		// if drop enabled, and revertible tx has error on commit, we skip the transaction and continue with next one
+		if algoConf.DropRevertibleTxOnErr && isRevertibleTx && hasCommitErr {
+			log.Warn("Found error on commit for revertible tx, but discard on err is enabled so skipping.",
+				"tx", txHash, "err", err)
+			continue
+		}
+
+		if hasCommitErr {
+			log.Trace("Bundle tx error", "bundle", bundle.OriginalBundle.Hash, "tx", txHash, "err", err)
 			return err
 		}
 
-		if receipt.Status != types.ReceiptStatusSuccessful && !bundle.OriginalBundle.RevertingHash(tx.Hash()) {
-			log.Trace("Bundle tx failed", "bundle", bundle.OriginalBundle.Hash, "tx", tx.Hash(), "err", err)
+		if hasReceiptFailed && !isRevertibleTx {
```

Comment:
From a readability perspective, I'd say putting `receipt != nil && receipt.Status == types.ReceiptStatusFailed` verbatim here is easier to read, since one has to jump around the code to see the definition of the variable.
Alternatively putting `hasReceiptFailed := receipt != nil && receipt.Status == types.ReceiptStatusFailed` right before this if would also be good

Reply:
updated